src/Button/index.js

```diff
@@ -1,365 +1,332 @@
+  font-weight: ${({ theme }) => theme.fontWeightBold};
+  font-size: ${({ tokens, size }) => tokens.fontSizeButton[size]};
+  cursor: pointer;
+  transition: all 0.15s ease-in-out;
+  outline: 0;
 
-  // iconSize in Button can be only small or medium
-  const sizeIcon = size === "large" || size === "normal" ? "medium" : size;
+  &:disabled {
+    cursor: default;
+    opacity: ${({ theme }) => theme.opacityButtonDisabled};
+  }
+  &:enabled:hover {
+    background: ${({ bordered, tokens, variation, type }) =>
+      bordered
+        ? tokens.backgroundButtonHover[variation]
+        : tokens.backgroundButtonHover[variation][type]};
+    border-color: ${({ bordered, tokens, variation, type }) =>
+      bordered && tokens.borderColorButtonHover[variation][type]};
+    color: ${({ tokens, variation, type }) => tokens.colorTextButtonHover[variation][type]};
+    & ${IconContainer} {
+      color: ${({ tokens, variation, type }) => tokens.colorTextButtonHover[variation][type]};
+    }
+  }
 
-  // allow onlyIcon only when Icon is passed
-  const allowOnlyIcon = !!(onlyIcon && Icon);
+  &:enabled:active {
+    transform: scale(${({ theme }) => theme.modifierScaleButtonActive});
+    background: ${({ bordered, tokens, variation, type }) =>
+      bordered
+        ? tokens.backgroundButtonActive[variation]
+        : tokens.backgroundButtonActive[variation][type]};
+    border-color: ${({ bordered, tokens, variation, type }) =>
+      bordered && tokens.borderColorButtonActive[variation][type]};
+    color: ${({ tokens, variation, type }) => tokens.colorTextButtonActive[variation][type]};
+    & ${IconContainer} {
+      color: ${({ tokens, variation, type }) => tokens.colorTextButtonActive[variation][type]};
+    }
+  }
+`;
+
+const Button = (props: Props) => {
+  const { children, bordered, size, icon, theme, type } = props;
+  const sizeIcon = size === "small" ? "small" : "medium";
+  const onlyIcon = icon && !children;
+  const variation = bordered ? "bordered" : "filled";
+  const tokens = {
+    heightButton: {
+      large: theme.heightButtonLarge,
+      normal: theme.heightButtonNormal,
+      small: theme.heightButtonSmall,
+    },
+    fontSizeButton: {
+      large: theme.fontSizeButtonLarge,
+      normal: theme.fontSizeButtonNormal,
+      small: theme.fontSizeButtonSmall,
+    },
+    paddingButton: {
+      large: theme.paddingButtonLarge,
+      normal: theme.paddingButtonNormal,
+      small: theme.paddingButtonSmall,
+    },
+    paddingButtonWithIcon: {
+      large: theme.paddingButtonLargeWithIcon,
+      normal: theme.paddingButtonNormalWithIcon,
+      small: theme.paddingButtonSmallWithIcon,
+    },
+    marginRightIcon: {
+      small: theme.marginButtonIconSmall,
+      normal: theme.marginButtonIconNormal,
+      large: theme.marginButtonIconLarge,
+    },
+    backgroundButton: {
+      filled: {
+        primary: theme.backgroundButtonPrimary,
```

Comment:
Can we please create constants for this ? like in `<Alert />`

Reply:
OK